go/arrow/csv/reader.go

```diff
@@ -721,6 +734,39 @@ func (r *Reader) parseDecimal256(field array.Builder, str string, prec, scale in
 	field.(*array.Decimal256Builder).Append(val)
 }
 
+func (r *Reader) parseList(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	if !(strings.HasPrefix(str, "{") || strings.HasSuffix(str, "}")) {
+		r.err = errors.New("invalid list format. should start with '{' and end with '}'")
+		field.AppendNull()
+		return
+	}
+	str = strings.TrimPrefix(str, "{")
+	str = strings.TrimSuffix(str, "}")
+	strs := strings.Split(str, ",")
+	if len(strs) == 0 {
+		field.AppendNull()
+		return
+	}
```

Comment:
an empty list is not equivalent to null. You should call `AppendEmpty` or you can call `Append(true)` and just not add anything to the value builder.